Userland/Libraries/LibJS/Runtime/ArrayConstructor.cpp

```diff
@@ -129,6 +136,243 @@ ThrowCompletionOr<NonnullGCPtr<Object>> ArrayConstructor::construct(FunctionObje
+    // 2. Let promiseCapability be ! NewPromiseCapability(%Promise%).
+    auto promise_capability = MUST(new_promise_capability(vm, realm.intrinsics().promise_constructor()));
+
+    // 3. Let fromAsyncClosure be a new Abstract Closure with no parameters that captures C, mapfn, and thisArg and performs the following steps when called:
+    SafeFunction<Completion()> from_async_closure = [constructor, mapfn, this_arg, &vm, &realm, async_items]() mutable -> Completion {
+        bool mapping;
+
+        // a. If mapfn is undefined, let mapping be false.
+        if (mapfn.is_undefined()) {
+            mapping = false;
+        }
+        // b. Else,
+        else {
+            // i. If IsCallable(mapfn) is false, throw a TypeError exception.
+            if (!mapfn.is_function())
+                return vm.throw_completion<TypeError>(ErrorType::NotAFunction, TRY_OR_THROW_OOM(vm, mapfn.to_string_without_side_effects()));
+
+            // ii. Let mapping be true.
+            mapping = true;
+        }
+
+        // c. Let usingAsyncIterator be ? GetMethod(asyncItems, @@asyncIterator).
+        auto using_async_iterator = TRY(async_items.get_method(vm, vm.well_known_symbol_async_iterator()));
+
+        GCPtr<FunctionObject> using_sync_iterator;
+
+        // d. If usingAsyncIterator is undefined, then
+        if (!using_async_iterator) {
+            // i. Let usingSyncIterator be ? GetMethod(asyncItems, @@iterator).
+            using_sync_iterator = TRY(async_items.get_method(vm, vm.well_known_symbol_iterator()));
+        }
+
+        GCPtr<Object> array;
+
+        // e. If IsConstructor(C) is true, then
+        if (constructor.is_constructor()) {
+            // i. Let A be ? Construct(C).
+            array = TRY(JS::construct(vm, constructor.as_function()));
+        }
+        // f. Else,
+        else {
+            // i. Let A be ! ArrayCreate(0).
+            array = MUST(Array::create(realm, 0));
+        }
+
+        // g. Let iteratorRecord be undefined.
+        Optional<IteratorRecord> iterator_record;
+
+        // h. If usingAsyncIterator is not undefined, then
+        if (using_async_iterator) {
+            // i. Set iteratorRecord to ? GetIterator(asyncItems, async, usingAsyncIterator).
+            iterator_record = TRY(get_iterator(vm, async_items, IteratorHint::Async, using_async_iterator));
+        }
+        // i. Else if usingSyncIterator is not undefined, then
+        else if (using_sync_iterator) {
+            // i. Set iteratorRecord to ? CreateAsyncFromSyncIterator(GetIterator(asyncItems, sync, usingSyncIterator)).
+            iterator_record = create_async_from_sync_iterator(vm, TRY(get_iterator(vm, async_items, IteratorHint::Sync, using_sync_iterator)));
+        }
+
+        // j. If iteratorRecord is not undefined, then
+        if (iterator_record.has_value()) {
+            // i. Let k be 0.
+            // ii. Repeat,
+            for (size_t k = 0;; ++k) {
+                // 1. If k ≥ 253 - 1, then
```

Comment:
`2^53`

Reply:
Woops! Manged that somehow with copy paste I guess, good spot!

Reply:
it's a very common issue, they use HTML `<sup>` in the spec so we have to manually turn that into something equivalent